AngularJS/components/reports/reportCRUD.js

```diff
@@ -0,0 +1,141 @@
+function ReportListController(reportService) {
+    //Lists all reports in tabular form, with add, edit and remove options against each report item.
+    var $ctrl = this;
+    $ctrl.master = {};
+    $ctrl.report = {};
+    $ctrl.isUpdate = false;
+    $ctrl.current_report_item = {};
+    $ctrl.types = [
+        {
+            keys: "AN",
+            value: "Annual"
+        }, 
+        {
+            keys: "MO",
+            value: "Monthly"
+        }, 
+        {
+            keys: "WE",
+            value: "Weekly"
+        }
+    ];
+
+    $ctrl.$onInit(){
+        get_token();
+        get_reports();
+    }
+
+    function get_token(){
+        reportService.get_token().then(
+        //gets token from service to authenticate.
+            function() {
+                console.warn();
+            },
+            function(errorMessage) {
+                console.warn(errorMessage);
+            }
+        );
+    }
+
+    function get_reports(){
+        reportService.get_reports().then(
+            list_reports,
+            function(errorMessage) {
+                console.warn(errorMessage);
+            }
+        );
+    }
+
+    function list_reports(response) {
+        //sets response data to list view (reports)
+        $ctrl.reports = response.data;
+    }
+
+    $ctrl.saveReportItem = function() {
+
+        if ($ctrl.isUpdate) {
+            updateReportItem();
+        }
+        else {
+            addReportItem();
+        }
+    }
+
+    function updateReportItem() {
+        /*updates report item by passing report_item as an argument, 
+        and sets new values of item in success.*/
+        report_item = current_report_item
+        report_item.title = $ctrl.report.title;
+        report_item.description = $ctrl.report.description;
+        reportService.update_report_item(report_item).then(
+            function update(response) {
+
+                if (response) {
+                    $ctrl.reports[getIndex(current_report_item.id)] = response.data;
+                    $ctrl.editForm = !$ctrl.editForm;
+                }
+            }
+        );
+    };
+
+    function addReportItem() {
+        /*Adds new report item by getting values from form fields,
+        and updates the list view in success.*/
+        reportService.add_report_item(get_form_data()).then(
+            function add(response) {
+
+                if (response.status === 201) {
+                    $ctrl.reports.push(response.data);
+                    $ctrl.showAddForm = false;
+                }
+            }
+        );
+    };
+
+    function get_form_data() {
+        //returns form fields values.
```

Comment:
Bindings help us by getting the value from form field to variable. Here you are not accessing form fields in any way.